source/fundamentals/crud/write-operations/embedded-arrays.txt

```diff
@@ -0,0 +1,188 @@
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you will learn how to update arrays in a document.
+
+To update an array, you must do the following:
+
+- Specify the update you want to perform
+- Specify what array elements to apply your update to
+- Perform an update operation using these specifications
+
+Sample Document
+~~~~~~~~~~~~~~~
+
+The following sections feature examples that update this sample
+document: 
+
+.. code-block:: json
+
+    { "_id": 1, "color": "green", "qty": [8, 12, 18] }
+
+The examples on this page use the ``findOneAndUpdate()`` method of the
+``MongoCollection`` class to retrieve and update the document. Each
+example uses an instance of the ``FindOneAndUpdateOptions`` class to
+have MongoDB retrieve the document after the update occurs. For
+more information on the ``findOneAndUpdate()`` method, see our Compound
+Operations guide <TODO>. 
+
+Specifying an Update
+--------------------
+
+To specify an update, use the ``Updates`` builder. The ``Updates``
+builder provides static utility methods to construct update
+specifications. For more information on using the ``Updates`` builder with
+arrays, see our :ref:`guide on the Updates builder <array_updates>`.
+
+The following example performs these actions: 
+
+- Query for the sample document
+- Add "17" to the ``qty`` array using the ``push()`` method of the ``Updates`` builder
+
+.. literalinclude:: /includes/fundamentals/code-snippets/UpdateArray.java
+   :language: java
+   :dedent:
+   :start-after: begin pushElementsExample
+   :end-before: end pushElementsExample
+
+The example above updates the original document to the following state:
+
+.. code-block:: json
+   :copyable: false
+
+   { "_id": 1, "color": "green", "qty": [8, 12, 18, 17] }
+
+Specifying Array Elements
+-------------------------
+
+You can specify which array elements to update using a positional
+operator. Positional operators can specify the first, all or certain
+array elements to update. 
+
+To specify elements in an array with positional operators, use **dot
+notation**. Dot notation is a property access syntax for navigating JSON
+objects. 
+
+For additional information, see the server manual entry on
+:manual:`dot notation </core/document/#std-label-document-dot-notation>`.
+
+The First Array Element
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To update the first element that matches your query filter, use the
+positional ``$`` operator. The array field must appear as part of your
+query document to use the positional ``$`` operator. 
+
+Example
+```````
+
+The following example performs these actions:
+
+- Query for a document with a ``qty`` field containing the value "18"
+- Decrement the first array value that matches the query document by "3"  using the ``inc()`` method of the ``Updates`` builder and the positional ``$`` operator 
+- Update and retrieve the document
```

Comment:
```suggestion
```

Suggestion: I believe this last bullet can be deleted on all examples where it is included (The First Array Element, All Array Elements, and Multiple Array Elements sections). The Specifying an Update section already does not include the third bullet, and I think it works well without it:

> Query for the sample document
Add "17" to the qty array using the push() method of the Updates builder

In addition, I think it is slightly confusing as I feel the verb (in this case decrement) implies that the value was changed. This sounds to me like 2 updates occured.